examples/interactive_local_hyperdrive_example.py

```diff
@@ -1,118 +1,228 @@
 
+# %% [markdown]
+#####################
+# Simulation Management
+#####################
 
+# %%
+# The local chain and hyperdrive objects provide various simulation tools, such as
+# advancing time, snapshotting, and controlling the underlying variable yield rate.
+
+# Advancing time.
 # NOTE these calls are chainwide calls, so all pools connected to this chain gets affected.
 # Advance time, accepts timedelta or seconds
 # The option `create_checkpoints` creates hyperdrive checkpoints when advancing time
 # but this call may be slow when advancing a large amount of time.
 chain.advance_time(datetime.timedelta(weeks=52), create_checkpoints=False)
 chain.advance_time(3600, create_checkpoints=True)
 
-# Close previous longs
-close_long_event_1 = agent0.close_long(
-    maturity_time=open_long_event_1.maturity_time, bonds=open_long_event_1.bond_amount
-)
+# Snapshotting.
+# Only one snapshot can be saved at a time. New snapshots made overwrite previous snapshots.
+chain.save_snapshot()
+chain.load_snapshot()
 
-agent0_longs = list(agent0.get_wallet().longs.values())
-close_long_event_2 = agent0.close_long(maturity_time=agent0_longs[0].maturity_time, bonds=agent0_longs[0].balance)
+# Set the underlying yield variable rate
+hyperdrive0.set_variable_rate(FixedPoint("0.10"))
 
-# Shorts
-open_short_event = hyperdrive_agent1.open_short(bonds=FixedPoint(33333))
-close_short_event = hyperdrive_agent1.close_short(
-    maturity_time=open_short_event.maturity_time, bonds=open_short_event.bond_amount
-)
+# %% [markdown]
+#####################
+# Analysis
+#####################
 
-# LP
-add_lp_event = hyperdrive_agent2.add_liquidity(base=FixedPoint(44444))
-remove_lp_event = hyperdrive_agent2.remove_liquidity(shares=hyperdrive_agent2.get_wallet().lp_tokens)
-
-# The above trades doesn't result in withdraw shares, but the function below allows you
-# to withdrawal shares from the pool.
-# withdraw_shares_event = hyperdrive_agent2.redeem_withdraw_share(shares=hyperdrive_agent2.wallet.withdraw_shares)
+# %%
+# Runs a dashboard showing various metrics on the pool.
+# This function automatically opens a browser tab with a dashboard.
+# Blocking mode waits for a keyboard press, then kills the dashboard server.
+# Non-blocking mode keeps the server running until cleanup.
+chain.run_dashboard(blocking=False)
 
 # %%
-# Get data from database under the hood
-pool_config = hyperdrive.get_pool_config()
-# The underlying data is in Decimal format, which is lossless. We don't care about precision
-# here, and pandas need a numerical float for plotting, so we coerce decimals to floats here
-pool_info = hyperdrive.get_pool_info(coerce_float=True)
+# These functions queries the underlying database to get data. All functions here return
+# a pandas dataframe.
 
-checkpoint_info = hyperdrive.get_checkpoint_info()
+# Agent level analysis
 
-# Change this to get wallet
-wallet = agent0.get_wallet()
+# Get the raw trade events in pandas across all pools
```

Comment:
You dropped using `.` at the end of your comments here. I personally think it's weird to require comments to be full sentences, but we do have that requirement in Solidity & Rust. More importantly, imo, is that we are consistent.

Reply:
(especially in the examples)